packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/util/CoroutineDispatcherFactory.kt

```diff
@@ -120,6 +120,10 @@ public class LiveRealmContext(
     }
 
     override fun close() {
+        // Warning: It is important to release the scheduler before closing the
```

Comment:
I guess this will work if we are guaranteed the `user_data_free` callback from core synchronously from `realm_release(scheduler)`. It would probably be safer for future changes with a separate abstraction of the scheduler that would set `dispatcherClosing` before `realm_release(scheduler)` ... but this is way better that the previous implementation that was relying on the message from the coroutine-exception 🙈